tests/test_io.py

```diff
@@ -8,86 +8,305 @@
+            [],
+            '.',
+            ['file.txt'],
+            'archive.zip',
+            'archive.zip',
+            '.',
+            None,
+            marks=pytest.mark.xfail(raises=FileNotFoundError),
+        ),
+        pytest.param(  # file not below root
+            ['file.txt', 'sub/'],
+            './sub',
+            ['../file.txt'],
+            'archive.zip',
+            'archive.zip',
+            '.',
+            None,
+            marks=pytest.mark.xfail(raises=ValueError),
+        ),
+        pytest.param(  # archive type not supported
+            [],
+            '.',
+            [],
+            'archive.bad',
+            'archive.zip',
+            '.',
+            None,
+            marks=pytest.mark.xfail(raises=RuntimeError),
+        ),
+        pytest.param(
+            ['archive.bad'],
+            '.',
+            [],
+            'archive.zip',
+            'archive.bad',
+            '.',
+            None,
+            marks=pytest.mark.xfail(raises=RuntimeError),
+        ),
+        pytest.param(  # broken archive
+            ['archive.zip'],
+            '.',
+            [],
+            'archive.tar.gz',
+            'archive.zip',
+            '.',
+            None,
+            marks=pytest.mark.xfail(raises=RuntimeError),
+        ),
+    ],
+    indirect=['tree'],
+)
+def test_archives(tmpdir, tree, root, files, archive_create,
+                  archive_extract, destination, expected):
+
+    root = audeer.path(tmpdir, root)
+    destination = audeer.path(tmpdir, destination)
+    archive_create = audeer.path(tmpdir, archive_create)
+    archive_extract = audeer.path(tmpdir, archive_extract)
+
+    # relative path
+
+    audeer.create_archive(
+        root,
+        files,
+        archive_create,
+    )
+    result = audeer.extract_archive(
+        archive_extract,
+        root,
+        keep_archive=True,
+    )
+    assert result == expected
+    assert os.path.exists(archive_extract)
+
+    # list of archives
+
+    result = audeer.extract_archives(
+        [archive_extract, archive_extract],
+        destination,
+        keep_archive=True,
+    )
+    assert result == expected + expected
+    assert os.path.exists(archive_extract)
+
+    # absolute path
+
+    if files is not None:
+        if isinstance(files, str):
+            files = audeer.path(root, files)
+        else:
+            files = [audeer.path(root, file) for file in files]
+        audeer.create_archive(
+            root,
+            files,
+            archive_create,
+        )
+        result = audeer.extract_archive(
+            archive_extract,
+            destination,
+            keep_archive=True,
+        )
+        assert result == expected
+        assert os.path.exists(archive_extract)
+
+    # delete archive
+
+    audeer.extract_archive(
+        archive_extract,
+        destination,
+        keep_archive=False,
+    )
+    assert not os.path.exists(archive_extract)
```

Comment:
Maybe also add a test that the extracted files exist at the expected place.

Reply:
done

Reply:
I changed to:

```python
    result = audeer.extract_archive(
        archive_extract,
        destination,
        keep_archive=False,
    )
    assert result == expected
    assert not os.path.exists(archive_extract)
```

but I guess that was not what you were asking for then :)

Reply:
Do you mean something like:

```
    assert result == expected
    for file in result:
        assert os.path.exists(audeer.path(root, file))
```

?

Reply:
Yes, exactly.